### PR TITLE
DOP-4048: responsive facet responses

### PR DESCRIPTION
**Pull request by seungpark**
[Pre-req PR](https://github.com/mongodb/docs-search-transport/pull/76)

[JIRA](https://jira.mongodb.org/browse/DOP-4048)

**Goal**: This PR is intended to update the functionality and format of our `/v2/search/meta` route, as to match the new format of the `/v2/status` route. This allows for front end to have dynamic facets with a single API query to the aforementioned route with query parameters for filters ([example](https://docs-search-transport.docs.staging.corp.mongodb.com/v2/search/meta?q=atlas&facets.target_product=atlas))

**Description:**
- `/v2/search/meta` route has been updated from a trie structure of shallow level of taxonomy, to a list structure as described in DOP-4048 for all depths. Front end can decide to show/hide based off the route response, without any data manipulation (with some special case logic of course..)

[Current meta search response](https://docs-search-transport.mongodb.com/v2/search/meta?q=atlas)
[Staged status response](https://docs-search-transport.docs.staging.corp.mongodb.com/v2/search/meta?q=atlas)

### CODE REVIEW

File: src/SearchIndex/util.ts [hunk 1]
```diff
@@ -31,77 +31,172 @@ interface FacetRes {
...
+function handleFacetValue(
+  facetOption: FacetOption,
+  key: string,
+  id: string,
+  taxonomyEntity: FacetDisplayNames,
+  facetsByFacetKey: { [key: string]: FacetOption | FacetValue }
+) {
+  facetOption.options.push({
+    id: id,
+    name: taxonomyEntity.name || convertTitleCase(taxonomyEntity.name || '', id),
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/SearchIndex/util.ts:54]: ah sorry this is bad var naming. will update the type name to be TrieFacet, not FacetDisplayNames. 

these have already been converted from taxonomy entity (with name and display_name) into a readable front end name

File: src/SearchIndex/util.ts [hunk 2]
```diff
@@ -31,77 +31,172 @@ interface FacetRes {
...
 ) {
-  let responseRef = responseFacets;
-  let taxRef = taxonomy;
-  try {
-    for (const facetName of facetKey.split('>')) {
-      const taxEntity = taxRef[facetName] as FacetDisplayNames;
-      if (!taxEntity) {
+  facetValue.facets.push({
+    id: id,
+    name: taxonomyEntity.name || convertTitleCase(taxonomyEntity.name || '', id),
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/SearchIndex/util.ts:72]: same as above. i dont think we need to convert here. [this has has already been done for the FacetDisplayNames type (renaming coming!) here](https://github.com/mongodb/docs-search-transport/pull/77/files#diff-08bf1d8bf4816ab5725baec42b163e6276eae0aaaff50d8eefe4c7318262f0dcR160-R169)

[the remark this answers is not in the document]
Comment (seungpark) [reply src/SearchIndex/util.ts]: updated! updated the typing name and we shouldn't need to convert here. the input is a `TrieFacet` (previosuly FacetDisplayNames 🤦 ) which already has been converted from TaxonomyEntity. no need to convert names here!

File: src/SearchIndex/util.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (seungpark) [reply src/SearchIndex/util.ts]: see [comment](https://github.com/mongodb/docs-search-transport/pull/77/files#r1341501562) below!

File: src/SearchIndex/index.ts [hunk 3]
```diff
@@ -21,8 +36,9 @@ export class SearchIndex {
   lastRefresh: RefreshInfo | null;
   documents: Collection<DatabaseDocument>;
   unindexable: Collection<DatabaseDocument>;
-  taxonomy: Taxonomy;
-  convertedTaxonomy: FacetOption[];
+  // taxonomy: Taxonomy;
```
[the remark this answers is not in the document]
Comment (seungpark) [reply src/SearchIndex/index.ts]: done!

**Review by rayangler — APPROVED** [review]

**Pull request merged**